### PR TITLE
[iOS] #735 - Allow keyboard update even if languageID not in API results

### PR DESCRIPTION
**Pull request by tombogle**

### CODE REVIEW

File: ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/KeyboardRepository.swift [hunk 1]
```diff
@@ -11,21 +11,22 @@ import Foundation
...
   public func installableKeyboard(withID keyboardID: String, languageID: String) -> InstallableKeyboard? {
-    guard let language = languages?[languageID] else {
-      return nil
-    }
-    guard let keyboard = language.keyboards?.first(where: { $0.id == keyboardID }) else {
+    guard let keyboard = keyboards?.first(where: { $0.key == keyboardID })?.value else {
       return nil
     }
+    
+    let language = keyboard.languages?.first(where: {$0.id == languageID}) ?? keyboard.languages?.first ?? languages?[languageID] ?? Language(name: languageID, id: languageID, keyboards: [keyboard], font: nil, oskFont: nil)
```
Comment (mcdurdin) [inline ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/KeyboardRepository.swift]: Is it possible to split this over multiple lines and comment on what is happening? It's a bit of a complex clause and a bit side-effecty :)

File: ios/keymanios.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>IDEDidComputeMac32BitWarning</key>
```
Comment (mcdurdin) [inline ios/keymanios.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Is this needed for ios ? Just a new artefact of XCode 9.3 maybe?

Comment (tombogle) [reply ios/keymanios.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Yes, it's a new artefact of XCode. It basically tells XCode that it's already done the check to see if the app is 32-bit in order to warn the developer that the clock is ticking on conversion of all code to 64-bit.

File: ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/KeyboardRepository.swift (hunk 1 above)
Comment (tombogle) [reply ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/KeyboardRepository.swift]: It is

Comment (tombogle) [reply ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/KeyboardRepository.swift]: @mcdurdin I have split this into four lines and am adding appropriate comments. The question now is whether we still need/want the fourth part? It' pretty kludgy and now will hopefully be unneeded. (But it does offer the ultimate safety net if something has gone wrong - which should help to ensure that we never find out about it and fix it.)

Comment (tombogle) [reply ios/engine/KMEI/KeymanEngine/Classes/KeyboardRepository/KeyboardRepository.swift]: I've removed the kludgy fallback. It's at least not needed to solve this issue and it's probably a bad idea.

**Pull request merged**